cmd/fs-v1.go

```diff
@@ -237,10 +237,16 @@ func (fs fsObjects) GetObject(bucket, object string, offset int64, length int64,
 		return toObjectErr(traceError(err), bucket, object)
 	}
 
+	// For offset '0' and negative length read everything.
+	if offset == 0 && length < 0 {
+		length = fi.Size
+	}
+
 	// Reply back invalid range if the input offset and length fall out of range.
 	if offset > fi.Size || length > fi.Size {
```

Comment:
We can remove the following `if offset+length > fi.Size {` block if we just modify this condition to `if offset > fi.Size || offset + length > fi.Size {`.

Also we seem to be missing a check for an error case where `offset > 0 && length < 0` before (or even in) this if block.